diff --git a/plugin/ref.js b/plugin/ref.js
index 5dba44ea5172515258806ff37f2da382c0235b61..987de1fdbece1b6c5b9c754cef4b4d41b21ae5fa 100644
--- a/plugin/ref.js
+++ b/plugin/ref.js
@@ -83,16 +83,23 @@ function overrideRef(t, path, refProp, metadata, state) {
     const binding = path.scope.getBinding(refProp.value.expression.name)
 
     // ref={ref}
-    if (isIdentifier && binding && t.isCallExpression(binding.path.node.init)) {
+    if (isIdentifier && binding && (t.isIdentifier(binding.path.node) || t.isCallExpression(binding.path.node.init))) {
         const userVariableName = refProp.value.expression.name
         const newRefFunction = t.arrowFunctionExpression(
             [t.identifier(uniqueRefName)],
             t.blockStatement([
                 t.expressionStatement(
-                    t.assignmentExpression(
-                        '=',
-                        t.memberExpression(t.identifier(userVariableName), t.identifier('current')),
-                        t.identifier(uniqueRefName)
+                    t.logicalExpression(
+                        '&&',
+                        t.identifier(userVariableName),
+                        t.assignmentExpression(
+                            '=',
+                            t.memberExpression(
+                                t.identifier(userVariableName), 
+                                t.identifier('current')
+                            ),
+                            t.identifier(uniqueRefName)
+                        )
                     )
                 ),
                 t.expressionStatement(
