diff --git a/node_modules/react-native-unistyles/.bun-tag-3fa499c76a95b00f b/.bun-tag-3fa499c76a95b00f
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/react-native-unistyles/.bun-tag-98acec6892fd75b3 b/.bun-tag-98acec6892fd75b3
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/plugin/ref.js b/plugin/ref.js
index 5dba44ea5172515258806ff37f2da382c0235b61..d9e0b7f144bc7c7e2231b48bb25c946de1a677d9 100644
--- a/plugin/ref.js
+++ b/plugin/ref.js
@@ -83,18 +83,37 @@ function overrideRef(t, path, refProp, metadata, state) {
     const binding = path.scope.getBinding(refProp.value.expression.name)
 
     // ref={ref}
-    if (isIdentifier && binding && t.isCallExpression(binding.path.node.init)) {
-        const userVariableName = refProp.value.expression.name
+    if (isIdentifier && binding && (t.isIdentifier(binding.path.node) || t.isCallExpression(binding.path.node.init))) {        const userVariableName = refProp.value.expression.name
         const newRefFunction = t.arrowFunctionExpression(
             [t.identifier(uniqueRefName)],
             t.blockStatement([
                 t.expressionStatement(
-                    t.assignmentExpression(
-                        '=',
-                        t.memberExpression(t.identifier(userVariableName), t.identifier('current')),
-                        t.identifier(uniqueRefName)
+                    t.logicalExpression(
+                        '&&',
+                        t.identifier(userVariableName),
+                        t.assignmentExpression(
+                            '=',
+                            t.memberExpression(
+                                t.identifier(userVariableName), 
+                                t.identifier('current')
+                            ),
+                            t.identifier(uniqueRefName)
+                        )
                     )
                 ),
+                // NOTE: This caused Uncaught SyntaxError: Invalid left-hand side in assignment
+                // t.expressionStatement(
+                //     t.assignmentExpression(
+                //         '=',
+                //         t.optionalMemberExpression(
+                //             t.identifier(userVariableName),
+                //             t.identifier('current'),
+                //             false,
+                //             true
+                //         ),
+                //         t.identifier(uniqueRefName)
+                //     )
+                // ),
                 t.expressionStatement(
                     t.callExpression(
                         t.memberExpression(t.identifier('UnistylesShadowRegistry'), t.identifier('add')),
